code/obj/item/pet_carrier.dm

```diff
@@ -0,0 +1,319 @@
+		else
+			boutput(user, "<span class='alert'>[src] is without any friends! Aww!</span>")
+		..()
+
+	// Ensure that things inside can actually breathe.
+	remove_air(amount)
+		var/turf/current_turf = get_turf(src)
+		. = current_turf.remove_air(amount)
+
+	return_air()
+		var/turf/current_turf = get_turf(src)
+		. = current_turf.return_air()
+
+	mob_flip_inside(mob/user)
+		..(user)
+
+		if (!src.can_break_out)
+			boutput(src, "<span class='alert'>It's no use! You can't leave [src]!</span>")
+			return
+		if (prob(BREAK_OUT_PROB))
+			if (length(src.carrier_occupants) > 1)
+				for (var/occupant in src.carrier_occupants)
+					src.eject_mob(occupant)
+				src.visible_message("<span class='alert'>[user] kicks the door of [src] open and OH GOD THEY'RE ALL ESCAPING!</span>")
+				return
+			src.eject_mob(user)
+			src.visible_message("<span class='alert'>[user] kicks the door of [src] open and crawls right out!</span>")
+			return
+		boutput(user, "<span class='alert'>Maybe this door could give out if you put up some more effort!</span>")
+
+	throw_impact(atom/hit_atom, datum/thrown_thing/thr)
+		..()
+		if (src.carrier_occupants)
+			for (var/mob/occupant in src.carrier_occupants)
+				occupant.throw_impact(hit_atom, thr)
+
+	/// Called when a given mob/user steals a mob after an actionbar.
+	proc/trap_mob(mob/mob_to_trap, mob/user)
+		if (!mob_to_trap)
+			return
+		if (mob_to_trap == user)
+			user.drop_item(src)
+		src.add_mob(mob_to_trap)
+		user.update_inhands()
+
+	/// Called when a given mob/user releases an mob after an actionbar.
+	proc/release_mob(mob/mob_to_release, mob/user)
+		if (mob_to_release)
+			src.eject_mob(mob_to_release)
+			user.update_inhands()
+			return
+		boutput(user, "<span class='alert'>Unable to release anyone from [src]!</span>")
+
+	/// Directly adds a target mob to the carrier.
+	proc/add_mob(mob/mob_to_add)
+		if (!mob_to_add)
+			return
+		mob_to_add.remove_pulling()
+		mob_to_add.set_loc(src)
+		mob_to_add.vis_flags |= src.mob_vis_flags
+		src.carrier_occupants.Add(mob_to_add)
+		src.vis_contents_proxy.vis_contents.Add(mob_to_add)
+		src.UpdateIcon()
+
+	/// Directly ejects a target mob from the carrier.
+	proc/eject_mob(mob/mob_to_eject)
+		if (!mob_to_eject)
+			return
+		MOVE_OUT_TO_TURF_SAFE(mob_to_eject, src)
+		mob_to_eject.vis_flags &= ~(src.mob_vis_flags)
+		src.carrier_occupants.Remove(mob_to_eject)
+		src.vis_contents_proxy.vis_contents.Remove(mob_to_eject)
+		src.UpdateIcon()
+
+	// There has to be a better way of checking that the other hand is empty. Please.
+	proc/return_other_hand_empty(mob/living/carbon/human/mob_to_check)
+		if (!ishuman(mob_to_check))
+			return FALSE
+		if (mob_to_check.hand && !mob_to_check.r_hand)
+			return TRUE
+		else if (!mob_to_check.l_hand)
+			return TRUE
+		else return FALSE
+
+	/// Calls src.AttackSelf(user) with a context action. Yeah, I know.
+	verb/release_occupant_verb(mob/user)
+		set name = "Release occupant"
+		set category = "Local"
+		set src in oview(1)
```

Comment:
Feel like oview might confuse some people who are standing right over it, it also makes the verb not able to be used when holding it in hand

Reply:
That last part's somewhat intentional, the verb's available for people who want to put the carrier down and subsequently open it up. If it's in your hand, you can already just attack_self(). As for oview, that's what a lot of other local verbs on larger non-item objects use, so that's what I've got.

Reply:
This verb doesn't seem to work? I'm guessing `AttackSelf` requires you to actually be holding it or something